device/core/src/pnp/command_request.ts

```diff
@@ -0,0 +1,36 @@
+import { JSONValue } from '.';
+import { DeviceMethodRequest } from '../device_method';
+
+export class CommandRequest {
+    /**
+     * The request identifier supplied by the service for this command call.
+     */
+    requestId: string;
+
+    /**
+     * The name of the command being called.
+     */
+    commandName: string;
+
+    /**
+     * The name of the component corresponding to the command being called.
+     */
+    componentName?: string;
+
+    /**
+     * The payload of this command call.
+     */
+    payload: JSONValue;
+
+    constructor(methodRequest: DeviceMethodRequest) {
+        this.requestId = methodRequest.requestId;
+        this.payload = methodRequest.payload;
+        const splitMethod = methodRequest.methodName.split('*');
+        if (splitMethod.length > 1) {
+            this.componentName = splitMethod[0];
+            this.commandName = splitMethod.slice(1).join('*');
```

Comment:
>splitMethod.slice(1).join('*'); [](http://example.com/codeflow?start=31&length=31)

`*` was chosen as a separator beause it's not valid for componentName, commandName, or methodName values.  You can juse ues `splitMethod[1]` here.

Reply:
Will the service fail the method/command request if an attempt is made to use `*` in the names? If not, I think we should either leave the code as it is now or throw when the user tries to register a listener with `*` in the name. Thoughts?